lib/swiftype-app-search/request.rb

```diff
@@ -86,6 +87,30 @@ def debug?
       @debug ||= (ENV['AS_DEBUG'] == 'true')
     end
 
+    def serialize_json(object)
+      JSON.generate(clean_json(object))
+    end
+
+    def clean_json(object)
+      case object
+      when Hash
+        object.transform_values { |value| clean_json(value) }
+      when Enumerable
+        object.map { |value| clean_json(value) }
+      else
+        clean_atom(object)
+      end
+    end
+
+    def clean_atom(atom)
+      case atom
+      when Time, DateTime
+        atom.iso8601
```

Comment:
I changed this to `to_datetime` and changed it to `is_a?` to support subclasses of `Time`.

I removed the check for `DateTime`, since it's unnecessary now.

I don't think we should `Date#to_datetime` since they're not really the same thing. `Date`s serialize fine anyways as strings.